pkg/ottl/functions.go

```diff
@@ -58,18 +58,18 @@ func (p *Parser[K]) buildArgs(inv invocation, fType reflect.Type) ([]reflect.Val
 	for i := 0; i < fType.NumIn(); i++ {
 		argType := fType.In(i)
 
-		switch argType.Kind() {
-		case reflect.Slice:
-			err := p.buildSliceArg(inv, argType, i, &args)
+		switch {
+		case argType.Kind() == reflect.Slice:
```

Comment:
The function reflection is complicated enough that I'd support turning this into an `if` to simplify things.  Same with `buildInternalArg`.  We can use switch statements once we get more conditions.

Tangentially, based on the discussion in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14712, I think we'll be including TelemetrySettings and the like in all functions, which should simplify newFunctionCall even further.

Reply:
To me, switch statements feel more straightforward in these places for a few reasons:

1. Switch statements express an intent that we're pivoting based on a single factor, in this case the type of an argument.
2. It makes it easier to append additional cases in the future if necessary. Otherwise someone either will add an if-else (which I would prefer to avoid if we have many cases) or need to change it to a switch statement. These are both in places where it's conceivable that we could add additional cases.
3. We're using switch statements for all other methods, so using switch statements adds some symmetry to the condition logic.

That said, I don't have a strong opinion on this, so I've changed them to if statements.